crates/dekoder/src/lib.rs

```diff
@@ -0,0 +1,172 @@
+//! eko output interface.
+use float_cmp::approx_eq;
+use ndarray::Array4;
+use std::collections::HashMap;
+use std::fs::remove_dir_all;
+use std::fs::File;
+use std::io::BufWriter;
+use std::path::PathBuf;
+use thiserror::Error;
+use yaml_rust2::Yaml;
+
+mod inventory;
+
+/// The EKO errors.
+#[derive(Error, Debug)]
+pub enum EKOError {
+    #[error("No working directory")]
+    NoWorkingDir,
+    #[error("I/O error")]
+    IOError(#[from] std::io::Error),
+    #[error("Loading operator from `{0}` failed")]
+    OperatorLoadError(PathBuf),
+    #[error("Failed to read key(s) `{0}`")]
+    KeyError(String),
+}
+
+/// My result type has always my errros.
+type Result<T> = std::result::Result<T, EKOError>;
+
+/// Default value for the float comparison inside `EvolutionPoint`.
+const EP_CMP_ULPS: i64 = 64;
+
+/// A reference point in the evolution atlas.
+pub struct EvolutionPoint {
+    /// Evolution scale.
+    pub scale: f64,
+    /// Number of flavors
+    pub nf: i64,
+}
+
+/// 4D evolution operator.
+pub struct Operator {
+    pub op: Option<Array4<f64>>,
+}
+
+impl Operator {
+    /// Empty initializer.
+    pub fn new() -> Self {
+        Self { op: None }
+    }
+}
+
+impl Default for Operator {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl TryFrom<&Yaml> for EvolutionPoint {
+    type Error = EKOError;
+
+    /// Load from yaml.
+    fn try_from(yml: &Yaml) -> Result<Self> {
+        // work around float representation
+        let scale = yml["scale"].as_f64();
+        let scale = if scale.is_some() {
+            scale.ok_or(EKOError::KeyError(
+                "because failed to read scale as float".to_owned(),
+            ))?
+        } else {
+            yml["scale"].as_i64().ok_or(EKOError::KeyError(
+                "because failed to read scale as float from int".to_owned(),
+            ))? as f64
+        };
+        let nf = yml["nf"]
+            .as_i64()
+            .ok_or(EKOError::KeyError("because failed to read nf".to_owned()))?;
+        Ok(Self { scale, nf })
```

Comment:
I would consider keeping using `serde-yaml` (death means it's completely stable :P), and then maybe transitioning to its successor (or directly `serde-json`) instead of going through manual parsing...

Reply:
Mmm not sure ...
a) as you can see above I already had to do some magic to make it work (because for an unknown reason in my test case the scale is saved as `10000` (note the missing `.`) and `as_f64` doesn't like that, so I don't know what `serde` would do
b) for the time being it is rather simple, so not sure we want to trade it for a dead crate
c) since we already planning (in the long/mid term) to replace it, I wonder if we can live with a non-optimal version for the moment

Reply:
> so I don't know what `serde` would do

The right thing, because `serde` is aware that the final type will be `f64` (is based on a schema, built by the `struct` definitions), so it will attempt a cast (or better, loading in the correct way from the beginning - I hope, though the specific loading is up to the deserializer implementation).

Instead, `yaml-rust2` is free-form, so it's inferring the type of the structure to create from the content itself. It could be handy for some applications, but not that much when the structure is known and fixed.

> b) for the time being it is rather simple, so not sure we want to trade it for a dead crate

The crate alive has potential to become quickly more dead than the previous one: it seems to be a fork of another one (https://github.com/chyh1990/yaml-rust) that died because of `serde-yaml` (the second contributor was the author of `serde-yaml`), and it has just been resurrected recently.
<img width="926" alt="image" src="https://github.com/user-attachments/assets/f05d9e5a-22f2-489d-96b4-43f371463190">

Being maintained is relevant for bug fixes, but being popular and used is telling you that there are fewer bugs around (and, if dead, no new one).

> c) since we already planning (in the long/mid term) to replace it, I wonder if we can live with a non-optimal version for the moment

Definitely. Mine is just an opinion, whose summary is something like: "they are both suboptimal, but one of the two is more painful than the other" xD

(I guess that in the end you'll want `serde` - but for the time being you can do whatever)

Reply:
> but for the time being you can do whatever

I would say let's reconsider the moment I start parsing the metadata (which I would not attempt in this PR)